src/main/java/henry/Parser.java

```diff
@@ -0,0 +1,121 @@
+    private static final Pattern DEADLINE_FORMAT = Pattern.compile("(?<desc>.+) /by "
+                                                                   + "(?<dateTime>\\d{2}-\\d{2}-\\d{4} "
+                                                                   + "\\d{2}:\\d{2})");
+
+    private static final Pattern EVENT_FORMAT = Pattern.compile("(?<desc>.+) /at "
+                                                                + "(?<dateTime>\\d{2}-\\d{2}-\\d{4} "
+                                                                + "\\d{2}:\\d{2})");
+    private static final String DATE_FORMATTER = "dd-MM-yyyy HH:mm";
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMATTER);
+
+    /**
+     * Parses a command from a string. There are two types of commands:
+     * single-argument commands, and double-argument commands. The function
+     * first checks if the command matches the appropriate pattern. If it does,
+     * then the corresponding command is returned. If not, then an exception is thrown.
+     *
+     * @param text the string to be parsed.
+     * @return the command corresponding to the string, or an exception if the string is malformed.
+     */
+    public Command parseCommand(String text) {
+        Matcher matcher = BASIC_COMMAND_FORMAT.matcher(text.trim());
+        if (!matcher.matches()) {
+            throw new HenryException("UNKNOWN COMMAND!");
+        }
+        String command = matcher.group("command").trim();
+        String args = matcher.group("args").trim();
+        switch (command) {
+        case EchoCommand.COMMAND_WORD:
+            return new EchoCommand(args);
+        case MarkCommand.COMMAND_WORD:
+            if (isArgsFormattedCorrectly(args)) {
+                throw new HenryException("ARGUMENT IS NOT A NUMBER!");
+            }
+            return new MarkCommand(Integer.parseInt(args));
+        case UnmarkCommand.COMMAND_WORD:
+            if (isArgsFormattedCorrectly(args)) {
+                throw new HenryException("ARGUMENT IS NOT A NUMBER!");
+            }
+            return new UnmarkCommand(Integer.parseInt(args.trim()));
+        case DeleteCommand.COMMAND_WORD:
+            if (isArgsFormattedCorrectly(args)) {
+                throw new HenryException("ARGUMENT IS NOT A NUMBER!");
+            }
+            return new DeleteCommand(Integer.parseInt(args.trim()));
+        case FindCommand.COMMAND_WORD:
+            return new FindCommand(parseFindArgs(args.trim()));
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case TodoCommand.COMMAND_WORD:
+            return new TodoCommand(args);
+        case DeadlineCommand.COMMAND_WORD:
+            return parseDeadlineArgs(args);
+        case EventCommand.COMMAND_WORD:
+            return parseEventArgs(args);
+        default:
+            throw new HenryException("UNKNOWN COMMAND!");
+        }
+    }
+
+    private String[] parseFindArgs(String args) {
```

Comment:
Missing JavaDocs for the next two methods

Reply:
Private methods need not have Javadoc comments; see my other response in the Henry.java class below.